source/Files.cpp

```diff
@@ -57,109 +57,123 @@ namespace {
 				c = '/';
 	}
 #endif
-}
-
-
-
-void Files::Init(const char * const *argv)
-{
-	// Parse the command line arguments to see if the user has specified
-	// different directories to use.
-	for(const char * const *it = argv + 1; *it; ++it)
-	{
-		string arg = *it;
-		if((arg == "-r" || arg == "--resources") && *++it)
-			resources = *it;
-		else if((arg == "-c" || arg == "--config") && *++it)
-			config = *it;
-			
-	}
 	
-	if(resources.empty())
+	void SetResourceDirectory(string &resources)
 	{
-		// Find the path to the resource directory. This will depend on the
-		// operating system, and can be overridden by a command line argument.
-		char *str = SDL_GetBasePath();
-		if(!str)
-			throw runtime_error("Unable to get path to resource directory!");
-		
-		resources = str;
-		SDL_free(str);
-	}
+		bool specifiedWithArg = resources.empty();
+		if(resources.empty())
+		{
+			// Find the path to the resource directory. This will depend on the
+			// operating system, and can be overridden by a command line argument.
+			char *str = SDL_GetBasePath();
+			if(!str)
+				throw runtime_error("Unable to get path to resource directory!");
+			
+			resources = str;
+			SDL_free(str);
+
+			// Create the "plugins" directory if it does not yet exist, so that it is
+			// clear to the user where plugins should go.
+			{
+				char *str = SDL_GetPrefPath("endless-sky", "plugins");
+				if(str != nullptr)
+					SDL_free(str);
+			}
+		}
```

Comment:
This should probably still happen even if the game is always started with a resource path.

(It seems more suited to config, as well)

Reply:
Are you thinking create plugins directory wherever config is set to (my preference) or always at this default location? re moving to config, whoops, yep will do.

Reply:
If a config directory is provided, then yes, we'd want to ensure the expected directory structure exists in it.

Reply:
Ok, so by default we'll use SDL_GetPrefPath("endless-sky", "plugins"); to create the directory, but if a custom config is specified we'll check for plugins and create a folder there if necessary using system specific calls.